Virtual cockpit

Jordan-64 opened the pull request:
There are several improvements in the VC, the most important being

Flashing has been added in the VC.
Panel 300-305 has been added in the CM.
Remodeled Circuit Breakers in both VC's.
New 3D model of the SPS Engine Injector Valves.
Added Hotkey (Left CTRL-H) for Arrows which pointing to the Cue Card Clickpoints.
Added simplified Interior of the White Room in LUT Service Arm 9.
Lighting switch in LM Ordeal is now working.
New thumbwheels in CMVC, which are now similar to the originals, have also been removed from the integrated lights, as the numbers on the wheels are black and therefore it makes no sense.
New Side Hatch in CMVC. Also the BPC has now a "Hatch" which is animated with the Side Hatch, so we can have a view in the White Room.
Added Crosspointer lights in LMVC
Added TTCA mesh in LM VC
The round dials in LM VC have been improved.
Improved more accurate round meters in LMVC.
Added Panel 230 in CMVC for all missions not functional for now until the work in 2D is done and merged.
Added working AOT reticle knob with display in LMVC

Jordan-64 committed:
Changes in CMVC
New ORDEAL animation. Now it moves back to the newly added Ordeal storage compartment.
The click points for the glare shades are now square instead of spherical and have the same area size as the meters or the DSKY

Changes in LMVC
The ORDEAL has been revised and is now like the one in the CMVC
Both FDAI crosses are now round corresponding to the FDAI sphere, previously they were flat.

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

Jordan-64 committed:
Fix: The wrong material was added to the Ordeal storage compartment.
I forgot to mention in my last commit that the arrows for the click point of the cue cards have been added. They are currently activated with the Flash switch in the Apollo Checklist MFD.

Jordan-64 committed:
Updated the ORDEAL Scaling animation.

Jordan-64 committed:
Added a hotkey (LEFT CTRL + H) to display the CueCardArrows in addition to the FLASH button in the ApolloCheckListMFD

Jordan-64 committed:
Changes in CM VC.
Added "Flashing" in VC. For now only in Command Module.
Revert back Panel 382 Cover to Clickmode Spherical do to Problems with Quadrilateral mode.
Made the Attitude Set Numbers in Panel 1 Round.
Disabled CueCard Pointing arrows Helper in PAMFD Flash, Can be enabled only CTRL+H

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
More circuit breakers and push buttons added to CMVC Flashing.

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

Jordan-64 committed:
New feature added to VC Flashing. A circle shows the position of the switch.

Jordan-64 committed:
Added VC Flashing in LM.
Removed unnecessary calls of UpdatePointingArrow()
Removed Old comments of CMVC Panel 5 and 100 Floodlights not working.
Changed the WasteDisposal animation using Parent animation instead of Combined.

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Added Panel 300-305 in CMVC.

Jordan-64 committed:
Fix Panel 300-302 Suit Flow Valves not Working. Big thanks to @TheRedMelon for the tip.

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
ADDED LUT Service arm 9 White Room Interior for not hiding the Side hatch when it is open.(The BoostCover needs to be Remodeled as well)
Reduced VC Flashing Pointing Arrow geometry.
VC Flashing Arrow is now rendered every frame for smoother animation.
Reduced Code for Waste Disposal Animation.
CM VC Thumb wheels remodeled for smoother animation.(UV's were distorted before)
Moved Code for Ordeal animation at last to prevent Switches floating away. (Unfortunately sometimes a few still floating away ant sometimes not)
Added SPS Injector Valves as 3D mesh instead as 2D, the need still to animated and the 2D code in VC needs to be removed. Initial code is there.
Added more Switches in CMVC Flashing. (Some of them still need special handling so that the arrow and the Circle points in the right spot)

rcflyinghokie committed:
Change sequencer lights to be component lights

Jordan-64 committed:
Merge pull request #18 from rcflyinghokie/VirtualCockpit

Change sequencer lights to be component lights

Jordan-64 committed:
Reworked Circuit Breakers in CM and LM VC
Connected the LM Ordeal to ordeal.LightingPower()
Reworked the Bags in LM VC

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

rcflyinghokie reviewed (dismissed):
#include <gcConst.h> does not allow building in a cmake/open orbiter build, the definitions need to be looked at/addressed.

This section should be removed from saturn.h unless there was a reason to include it, in which case it could be moved to saturn.cpp

Added a PR to this branch that removes the code.

rcflyinghokie committed:
Remove ifdef code that breaks build

Jordan-64 committed:
Fixed LM VC Flashing arrow that didn't update every frame.
Improved 3D mesh for the Direct O2 Rotary in Panel 7 and others similar.

Jordan-64 committed:
Merge pull request #19 from rcflyinghokie/VirtualCockpit

Remove ifdef code that breaks build

Jordan-64 committed:
CMVC changes.
New mesh for the PostLanding Vent Handle, the Direct O2 Rotary in Panel 7, and similar ones.
Some geometries have been reduced so they don't exceed the 50 MB GitHub complaints. The visual experience should not be impaired.
Fix Flashig arrow not pointing at Forward Hatch Equal Valve.

Jordan-64 committed:
Fixed flashing arrow for the thumbwheels pointing behind the hull in CMVC.
CODE IS UGLY BUT IT WORKS.

Jordan-64 committed:
The code for switches that require extra handling to make the flashing arrow point to the correct position has been rewritten.
New thumbwheels in CMVC, which are now similar to the originals, have also been removed from the integrated lights, as the numbers on the wheels are black and therefore it makes no sense.

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
FIX LMVC DSKY Some cells were lit in missions before 11 when they shouldn't.
FIX CMVC Abort Light disappears when switching in Tunnel view.
CMVM better Texture for the Thumbwheels.

Jordan-64 committed:
Reactivation of the temporarily deactivated draw function of the talkbacks.

Jordan-64 committed:
Reduced Poly count in CMVC mesh from about 47+ MB to 36+ MB mostly from the Switch body which are not noticeably.
Also renamed some names in Switchparts from SwitchHolder to SwitchBody and VC_MAT_HolderRahmen_t to VC_MAT_SwitchGuard_t, which makes more sense

Jordan-64 committed:
Fix: The CueCard Arrow shortcut blocks other shortcuts, such as NASSP's own quicksave.

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
A new side hatch has been added to the CMVC. Unfortunately, it's not possible to make it with just one texture, as the hatch has a lot of details and therefore requires more texture space. I think that will come later.
I gave the SPS valve needles a transparent texture so that they are invisible until someone can animate them.

Jordan-64 committed:
Fix, the LMVC pointing arrow does not point to the Forward Hatch Relief Valve Lever.

Jordan-64 committed:
A "hatch" has been added to the BPC that is hidden when the side hatch is open, so that when you are in the launch pad and the hatch is open you have a clear view of the white room.
Other changes, some small code and mesh cleanup.

indy91 committed:
Implement the SPS engine injector indicators as round meters

Jordan-64 committed:
Merge pull request #20 from indy91/VirtualCockpit

Implement the SPS engine injector indicators as round meters

indy91 committed:
Delete more code related to the now deleted SPS injector valve indicator bitmap

Jordan-64 committed:
Merge pull request #21 from indy91/VirtualCockpit

Delete more code related to the now deleted SPS injector valve indicator bitmap

Jordan-64 committed:
Fix wrong Pivot point in CMVC Side Hatch Gear Box Selector
Fix BPC not Hidden when starting in PAUSE mode
Added Crosspointer lights in LMVC

Jordan-64 committed:
FIX VC Flashing does not point at the correct valves in LM VC Upper and Forward Hatch

Jordan-64 committed:
FIX Mission Timer bug in LMVC.dds

Jordan-64 committed:
Fix. The flashing arrow in the LM VC pointed to the wrong position.

indy91 committed:
CSM ORDEAL is stowed by default; add protection against missing DefineMeshGroup calls; remove all special cases for VC checklist aid (arrow and circle)

indy91 committed:
CSM side and forward hatch have proper handling of clickspots and VC checklist helper (arrow and circle)

Jordan-64 committed:
Merge pull request #22 from indy91/VirtualCockpit

Remove special cases for VC checklist aid (arrow and circle) and other VC code improvements

indy91 committed:
Fix CSM side hatch actuator selector handle clickspot when the hatch has just been closed

Jordan-64 committed:
Merge pull request #23 from indy91/VirtualCockpit

Fix CSM side hatch actuator selector handle clickspot when the hatch …

Jordan-64 committed:
Added TTCA mesh in LM VC

indy91 committed:
VC update code cleanup; move SetVCCueCardsArrows call to clbkVCRedrawEvent; only check the key combination for viewing cue cards (LCTRL + H) when in the VC, thereby not blocking the 2D panel HUD key combination

Jordan-64 committed:
Merge pull request #24 from indy91/VirtualCockpit

VC update code cleanup; move SetVCCueCardsArrows call to clbkVCRedraw…

Jordan-64 committed:
Update csm_main_panel.bmp

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Some Modifications in LM TTCA to avoid mesh intersection with the Armrests

indy91 committed:
Make initial animation state of RoundMeter configurable; delete unnecessary LM Steerable S-Band antenna meter code

indy91 committed:
Fix typo

Jordan-64 committed:
Merge pull request #25 from indy91/VirtualCockpit

Make initial animation state of RoundMeter configurable; delete unnecessary LM Steerable S-Band antenna meter code

Jordan-64 committed:
New Meters in LM Panel 12 and 14

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Update LM VC Meters

Jordan-64 committed:
Add Window 2 in BCP mesh

indy91 committed:
Add animation of boost protective cover hatch

Jordan-64 committed:
Merge pull request #26 from indy91/VirtualCockpit

Add animation of boost protective cover hatch

Jordan-64 committed:
FIX flashing arrow of the LMVC AbortSwitch points in the wrong position.
Made the backgrounds of the round dials in LM VC darker.

Jordan-64 committed:
FIX The text on LMVC panel 14 is "DC BUS FAULT" instead of "DC FEEDER FAULT".

Jordan-64 committed:
Added Inverter and Recorder guards in LMVC Panel 12 and 14
DC FEEDER FAULT text in LMVC panel 14 depends on the mission flown.

Jordan-64 committed:
Added U L N text in CMVC Sidehatch Actuator Handle Texture
Renamed function ChangeMeshGroupVisibility() to HideMeshGroup() which is more meaningful

Jordan-64 committed:
FIX The needles of the SPS valve had a transparent texture.

Jordan-64 committed:
FIX CMVC HighGainAntenna PitchMeter not Displaying the Correct Angle.

Jordan-64 committed:
New in this update...

CMVC
The rotarys and text in panels 251 and 252 have been improved. This should fix bug https://github.com/orbiternassp/NASSP/issues/1341#event-20318298108.
The white semicircle in both FDAIs now has a "support." This should fix bug https://github.com/orbiternassp/NASSP/issues/1351#event-20318294910.

LMVC
The white lines on the panels have been refined.

CMVC and LMVC
The checklist circle is now animated for better visibility.

rcflyinghokie reviewed (commented):
These changes cause lengthy hanging on the loading screen, need to be fixed

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into VirtualCockpit

rcflyinghokie committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP.git into VirtualCockpit

Jordan-64 committed:
Some debugging code

rcflyinghokie committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP.git into VirtualCockpit

indy91 committed:
Move HideMeshGroup calls to better places; fix a few switch/case bugs

Jordan-64 committed:
Merge pull request #27 from indy91/VirtualCockpit

Move HideMeshGroup calls to better places; fix a few switch/case bugs

rcflyinghokie reviewed (commented):
Seems to fix the hang on my custom flight

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Updatedet round meters in CMVC Panel 2
Improved text in CMVC LEB area

Jordan-64 committed:
New improved O2 flow gauge in CMVC Panel 2

Jordan-64 committed:
Add missing White Room Textures

Jordan-64 committed:
Fix EcsRadTempInletMeter Pivot point in CMVC
Fix some Gauge visuals in CMVC mesh

Jordan-64 committed:
FIX csm_right_cb_panels.dds texture contains testing parts of panel 230 that have not yet been implemented in VC

Jordan-64 committed:
Add missing LM VC hose texture

Jordan-64 committed:
Move all the variables in Checklist Arrow on top of the Function and make them static

Jordan-64 committed:
Add Panel 230 in CMVC. Non functional for now

Jordan-64 committed:
Added Apollo 17 Panel 230 in CMVC, shortened and added the switch positions in code.

Jordan-64 committed:
Remove unnecessary "CMVC_Panel230.dds" textur. Is replaced with "CMVCPanel230_A15.dds"

Jordan-64 committed:
Fix Text in CMVC He TK TEMP Meter, it says "Hq TK" instead of "% °F"

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Added Integral Texture for CMVC Panel 230
Combined Texture for CMVC Telescope and Sextant in One and removed their Separated textures.
Added new Texture for the CMVC TB's and adjusted the Opening and closing speed, which also changes the TB's speed in LM
New AOT ReticleKnob mesh, with animation (NOT LINKED WITH THE KNOB FROM 2D FOR NOW)
Added working AOT Reticle Display in CMVC

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

Jordan-64 committed:
Added missing EMS Texture

Jordan-64 committed:
The AOT reticle knob now works as expected in the virtual cockpit. The problem is that the knob animation is not updated when it is operated in 2D mode.

Jordan-64 committed:
Fix AOT Reticle Knob not updating in VC while operating in 2D

rcflyinghokie committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP.git into VirtualCockpit

rcflyinghokie committed:
update power/signal failure light off color

Jordan-64 committed:
Merge pull request #28 from rcflyinghokie/VirtualCockpit

Update power/signal failure light off color

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

indy91 committed:
Convert UpdatePointingArrow function to a common class for CSM and LM

Jordan-64 committed:
Merge pull request #29 from indy91/VirtualCockpit

Convert UpdatePointingArrow function to a common class for CSM and LM

Jordan-64 committed:
FIX MERGE CONFLICTS
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into VirtualCockpit

Jordan-64 committed:
FIX for Floating away switches in ORDEAL Animation. I do now manipulate the vertices directly instead of doing a "combined" animation

rcflyinghokie reviewed (dismissed):
Has been tested in O16 and O24 pretty extensively

rcflyinghokie committed:
Add combined output to "ambient light" code

Jordan-64 committed:
Merge pull request #30 from rcflyinghokie/VirtualCockpit

Add combined output to "ambient light" code

Jordan-64 committed:
Small Code Update for the ORDEAL Animation

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit
